src/apps/competitions/tests/test_phase_migration.py

```diff
@@ -0,0 +1,98 @@
+import datetime
+import time
+
+from django.test import TestCase
+from django.contrib.auth import get_user_model
+from django.utils.timezone import now
+from rest_framework.test import APIClient
+
+from competitions.tasks import do_phase_migrations
+from factories import UserFactory, CompetitionFactory, PhaseFactory, SubmissionFactory, SubmissionScoreFactory, \
+    CompetitionParticipantFactory
+
+User = get_user_model()
+
+
+class CompetitionPhaseToPhase(TestCase):
+    def setUp(self):
+        self.owner = UserFactory(username='owner', super_user=True)
+        self.normal_user = UserFactory(username='norm')
+        self.normal_user2 = UserFactory(username='norm2')
+        self.normal_user3 = UserFactory(username='norm3')
+        self.competition = CompetitionFactory(created_by=self.owner)
+        self.competition_participant = CompetitionParticipantFactory(user=self.normal_user,
+                                                                     competition=self.competition)
+        self.competition_participant2 = CompetitionParticipantFactory(user=self.normal_user2,
+                                                                      competition=self.competition)
+        self.competition_participant3 = CompetitionParticipantFactory(user=self.normal_user3,
+                                                                      competition=self.competition)
+        self.phase1 = PhaseFactory(competition=self.competition, auto_migration=True, status='Current')
+        self.phase2 = PhaseFactory(competition=self.competition, auto_migration=True, status='Next')
+        self.phase3 = PhaseFactory(competition=self.competition, auto_migration=False, status='Final')
+        self.submission = self.make_submission()
+        self.submission2 = self.make_submission(participant=self.competition_participant2)
+        self.client = APIClient()
+
+    def make_submission(self, **kwargs):
+        kwargs.setdefault('owner', self.owner)
+        kwargs.setdefault('participant', self.competition_participant)
+        kwargs.setdefault('phase', self.phase1)
+        kwargs.setdefault('status', 'None')
+        sub = SubmissionFactory(**kwargs)
+        SubmissionScoreFactory(submission=sub)
+        return sub
+
+    def test_migrate_submissions(self):
+        """
+        Determine whether all submissions in phase1 end up in phase2
+        :return:
```

Comment:
remove the returns here, I believe proper style is to do docstrings like so:

```python
"""I'm on the same line as the quotes!"""
```

..but, I think these are probably unnecessary. Test names should be of the form: `test_<what i am testing>_<what i expect>` like `test_migrating_submissions_moves_submissions_to_next_phase`

Reply:
Cruft removed in latest push and made test names clearer for their objective.